src/serde_support.rs

```diff
@@ -0,0 +1,266 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+extern crate alloc;
+
+use alloc::format;
+use alloc::string::String;
+
+use dusk_bytes::Serializable;
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
```

Comment:
You use `de::Error` and `serde::de::Error` a lot so I think it makes sense to import and alias it:
```suggestion
use serde::{Deserialize, Deserializer, Serialize, Serializer};
use serde::de::Error as SerdeError;
```